### PR TITLE
sync Ray Tune search space and docs with native tuner ranges

### LINKED ISSUE #23926: Hyperparameters for Yolov26
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### ❓ Questions on Loss Functions & Hyperparameter Tuning in YOLOv26

Hi team,

I have a couple of questions and some confusion regarding loss functions and hyperparameter tuning in YOLOv26.

---

### 🔹 Q1: Box Loss Gain — Default vs Tuning Range

In the official hyperparameter tuning documentation, I see the following suggested range:

```yaml
box: tune.uniform(0.02, 0.2)
```

However, in the default config file, the value is:

```yaml
box: 7.5  # (float) box loss gain
```

This seems inconsistent.

Additionally, when I run hyperparameter tuning (using genetic algorithm) starting from the default value of `7.5`, the optimized values I get are typically around `5–8`, which are far from the recommended tuning range (`≤ 0.2`) and when I reduce it to 0.2 let's say I always get 20-30% lower accuracy than I would get from a value of 5 but official docs upper limit is 0.2 so that doesn't add up.

**Questions:**

* Why is the default value (`7.5`) so far from the suggested tuning range (`0.02–0.2`)?
* Is the tuning range outdated, or does it apply to a different scaling/normalization setup?
* Should tuning ideally start from the lower range instead of the default config?

---

### 🔹 Q2: DFL Loss Behavior in YOLOv26

From what I understand, `dfl_loss` has been removed or is no longer actively used in YOLOv26 training.

However, during training, I still observe `dfl_loss` being logged and even optimized as part of the training metrics.

**Questions:**

* Is `dfl_loss` still being computed internally but not used in backpropagation?
* If it is not used, why is it still being tracked and optimized?
* Is this expected behavior or a leftover artifact from previous versions?

---

### 🔹 Environment

* `ultralytics`: 8.4.3
* `torch`: 2.7.0+cu128
* `python`: 3.10.13

### Additional

_No response_

### PR DESCRIPTION
**Pull request by raimbekovm**
## Summary

Fixes #23926

The Ray Tune default search space (`utils/tuner.py`) and both hyperparameter tuning docs pages had outdated ranges from the YOLOv8 era that were never synced when the native tuner was updated for YOLO26 in PR #23176 (commit 06c41e675, Dec 2025).

Most critically, `box` loss range was `(0.02, 0.2)` while the default value is `7.5` and the correct native range is `(1.0, 20.0)` — as reported by the user, tuning with the old range produced 20-30% lower accuracy than values around `5-8` that the native tuner finds.

### Changes

**Code** (`ultralytics/utils/tuner.py`):
- Synced all 7 divergent ranges with native tuner: `lr0`, `momentum`, `box`, `cls`, `scale`
- Added 2 missing params: `dfl` and `close_mosaic`

**Docs** (`docs/en/guides/hyperparameter-tuning.md`):
- Updated table with correct ranges for `lr0`, `momentum`, `box`, `cls`, `dfl`, `scale`
- Added missing `cutmix` param
- Fixed `close_mosaic` type from `int` to `float`
- Updated code examples

**Docs** (`docs/en/integrations/ray-tune.md`):
- Updated both tables (main + FAQ) with correct ranges
- Added missing params: `dfl`, `bgr`, `close_mosaic`
- Updated code examples and descriptions

**Docstring** (`ultralytics/engine/tuner.py`):
- Updated example in class docstring

### Full diff (old → new)

| Param | Old (RayTune/Docs) | New (Native) | Default |
|---|---|---|---|
| `lr0` | `(1e-5, 1e-1)` | `(1e-5, 1e-2)` | `0.01` |
| `momentum` | `(0.6, 0.98)` | `(0.7, 0.98)` | `0.937` |
| `box` | `(0.02, 0.2)` | `(1.0, 20.0)` | `7.5` |
| `cls` | `(0.2, 4.0)` | `(0.1, 4.0)` | `0.5` |
| `dfl` | missing | `(0.4, 12.0)` | `1.5` |
| `scale` | `(0.0, 0.9)` | `(0.0, 0.95)` | `0.5` |
| `close_mosaic` | missing in RayTune | `(0.0, 10.0)` | `10` |

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
This PR aligns Ray Tune hyperparameter search spaces and related documentation with the native Ultralytics tuner defaults for more accurate, consistent tuning guidance. 🚀

### 📊 Key Changes
- Updated Ray Tune default search ranges in `ultralytics/utils/tuner.py` to match native tuner values, including changes to `lr0`, `momentum`, `box`, `cls`, `dfl`, `scale`, and `close_mosaic`.
- Added missing parameters to Ray Tune coverage and docs, including `dfl`, `bgr`, and `close_mosaic`, while keeping `cutmix` documented consistently.
- Synced documentation in `docs/en/guides/hyperparameter-tuning.md` and `docs/en/integrations/ray-tune.md` with the current tuner ranges and parameter types.
- Updated code examples and inline docstrings in `ultralytics/engine/tuner.py` and docs to use the corrected ranges, especially `lr0` and `momentum`.
- Adjusted `close_mosaic` documentation from `int` to `float` to reflect the actual tuning space representation.

### 🎯 Purpose & Impact
- Reduces confusion for users by ensuring docs, examples, and Ray Tune defaults all reflect the same tuning behavior. ✅
- Improves reliability of hyperparameter tuning workflows by exposing the intended native search ranges in Ray Tune. 🎯
- Helps users avoid overly broad or outdated search intervals that could waste tuning trials or produce inconsistent results. ⚡
- Strengthens maintainability by keeping implementation and documentation synchronized across tuning entry points. 📚

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness; an Ultralytics engineer will assist you soon. For this docs + tuner sync update, please review the checklist below to help ensure a smooth merge. 😊

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

File: ultralytics/utils/tuner.py [hunk 1]
```diff
@@ -78,6 +79,7 @@ def run_ray_tune(
         "mixup": tune.uniform(0.0, 1.0),  # image mixup (probability)
         "cutmix": tune.uniform(0.0, 1.0),  # image cutmix (probability)
         "copy_paste": tune.uniform(0.0, 1.0),  # segment copy-paste (probability)
+        "close_mosaic": tune.uniform(0.0, 10.0),  # close dataloader mosaic (epochs)
```
Comment (UltralyticsAssistant) [inline ultralytics/utils/tuner.py:82]: 💡 **MEDIUM**: `bgr` is now documented in both tuning guides as part of the default Ray Tune space, but it is still missing from `default_space` here. That leaves the implementation out of sync with the docs and with the PR summary, and users relying on the documented defaults won't actually tune `bgr` unless they add it manually.

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The PR mostly does what it says: the Ray Tune ranges and docs are updated consistently for `lr0`, `momentum`, `box`, `cls`, `dfl`, `scale`, and `close_mosaic`. The main issue I found is that `bgr` was added to the documentation but not to the actual Ray Tune `default_space`, so the implementation is still out of sync with the published defaults.

💬 Posted 1 inline comment

File: ultralytics/utils/tuner.py (hunk 1 above)
Comment (Y-T-G) [reply ultralytics/utils/tuner.py:82]: @raimbekovm Can you fix this?

Comment (raimbekovm) [reply ultralytics/utils/tuner.py:82]: i think this was resolved after merging with main — `bgr` was already added upstream in 2ee0b4d8

**Review by Y-T-G — APPROVED** [review]

**Comment by glenn-jocher** [comment]
Merged, thanks for syncing `utils/tuner.py` and the docs so Ray Tune now matches the native tuner defaults; the updated [Ray Tune integration guide](https://docs.ultralytics.com/integrations/ray-tune) and [hyperparameter tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning) now reflect the corrected ranges on `main`.
